common/djangoapps/pipeline_mako/templates/mako/inline_js.html

```diff
@@ -1,9 +1,12 @@
+<%page expression_filter="h"/>
+<%! from openedx.core.djangolib.js_utils import dump_js_escaped_json %>
 <script \
-% if async:
+% if context.get('async', default=None):
 async \
 % endif
-if defer:
+% if defer:
 defer \
+% endif
 type="text/javascript" charset="utf-8">
-    ${source | safe}
+    ${source | n, dump_js_escaped_json}
```

Comment:
Curious if you were able to test and ensure this doesn't double-escape?

Reply:
Also, if the linter keeps complaining because it thinks this is an html context, you may need to disable the linter on this line.

Reply:
I have not been able to test this. First of all I think that this is not being used in the platform, you can see the syntax error that this file had in the lines above that I changed.

As far as I see, this file is not used in production because this only can be used when the [PIPELINE_ENABLED](https://github.com/edx/edx-platform/blob/083f1bb8b2ed6f84e47a927288adeb6c4de8d9f3/common/djangoapps/pipeline_mako/__init__.py#L57) is false. 

I tried to use that in the devstack but it seems to me that it is not used there because of this https://github.com/edx/edx-platform/blob/083f1bb8b2ed6f84e47a927288adeb6c4de8d9f3/common/djangoapps/pipeline_mako/templates/static_content.html#L67, compress_js is never called.

I tried to trace where the value of source comes from, and it comes from this: https://github.com/edx/edx-platform/blob/083f1bb8b2ed6f84e47a927288adeb6c4de8d9f3/common/djangoapps/pipeline_mako/__init__.py#L59, in all the tests that I made in the devstack(after a change to the static-contents.html), that is empty.

That is from [django-pipeline](https://github.com/jazzband/django-pipeline/blob/07cfee8ba6ea4c87cd0d9c15e76c7f6d79cf981e/pipeline/packager.py#L124) dependency, as far as I have seen there, compile_templates does not escape the file. So I think that it will not get double escaped.

They, in the jinja version of this, use the safe filter for that https://github.com/jazzband/django-pipeline/blob/07cfee8ba6ea4c87cd0d9c15e76c7f6d79cf981e/pipeline/jinja2/pipeline/inline_js.jinja

Reply:
@morenol: 
1. I think the changes you have done should work.
2. The `safe` tag does not escape.  It is like using out `HTML()` function in Mako.  It just marks it as safe so future escaping won't be done. See https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#safe
3. @dianakhuang: Maybe you can consider this for DEPR?  It is sad that so much maintenance work needs to go into something that possibly could be removed.

Reply:
Is this a DEPR for `dump_js_escaped_json`? Sorry, I'm not caught up on this conversation yet.

Reply:
@dianakhuang: Sorry.  No.  A DEPR for this template.